src/viking.js

```diff
@@ -1,11 +1,109 @@
-// Soldier
-function Soldier() {}
+// Soldier ++++++++++++++++++++++++++++++++++++++++++++++++++
+function Soldier (healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg
+}
 
-// Viking
-function Viking() {}
+Soldier.prototype.attack = function() {
+    return this.strength
+}
 
-// Saxon
-function Saxon() {}
+Soldier.prototype.receiveDamage = function(damage) {
+    this.health -= damage;
+}
 
-// War
-function War() {}
+// Viking ++++++++++++++++++++++++++++++++++++++++++++++++++
+function Viking(name, healthArg, strengthArg) {
+    Soldier.call(this, healthArg, strengthArg);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype); 
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function(damage) {
+    this.health -= damage;
+    if (this.health <= 0) {
+        return `${this.name} has died in act of combat`;
+    } else {
+        return `${this.name} has received ${damage} points of damage`;
+    }
+}
+
+Viking.prototype.battleCry = function (){
+    return "Odin Owns You All!"
+}
+
+
+// Saxon ++++++++++++++++++++++++++++++++++++++++++++++++++
+function Saxon(healthArg, StrengthArg) {
+    this.health = healthArg;
+    this.strength = StrengthArg;
+}
+
+Saxon.prototype.attack = function(){
+    return this.strength;
+}
+
+Saxon.prototype.receiveDamage = function(damage) {
+    this.health -= damage;
+    if (this.health <= 0) {
+        return `A Saxon has died in combat`
+    } else {
+        return `A Saxon has received ${damage} points of damage`
+    }
+}
+
+Saxon.prototype = Object.create(Soldier.prototype); 
+Saxon.prototype.constructor = Saxon;
+
+// War ++++++++++++++++++++++++++++++++++++++++++++++++++
+function War () {
+    this.vikingArmy = [];
+    this.saxonArmy =[];
+}
+
+War.prototype.addViking = function(Viking){
+    this.vikingArmy.push(Viking)
+}
```

Comment:
Here, the argument "viking" passed to the function addViking() should be lower-case (viking and not Viking), as it is a placeholder for any soldier of the viking army (not to be confused with the constructor function Viking() )